### PR TITLE
Rename config.toml to bootstrap.toml

### PR DESCRIPTION
**Pull request by LuuuXXX**
### What does this PR try to resolve?

Cargo and bootstrap use the same name for their configuration file. this is confusing. 

This pr attempts to disambiguate the naming, makes it easier to understand.

cc https://github.com/rust-lang/rust/issues/126875

note: config.toml is still supported, so this doesn't break the downstreams.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
r? bootstrap

File: src/bootstrap/src/utils/change_tracker.rs [hunk 1]
```diff
@@ -173,7 +173,7 @@ pub const CONFIG_CHANGE_HISTORY: &[ChangeInfo] = &[
     ChangeInfo {
         change_id: 123711,
         severity: ChangeSeverity::Warning,
-        summary: "The deprecated field `changelog-seen` has been removed. Using that field in `config.toml` from now on will result in breakage.",
+        summary: "The deprecated field `changelog-seen` has been removed. Using that field in `bootstrap.toml` from now on will result in breakage.",
```
Comment (onur-ozkan) [inline src/bootstrap/src/utils/change_tracker.rs:176]: We should revert the current changes in `change_tracker.rs` and add a new change entry to inform people about this renaming work.

**Review by onur-ozkan — COMMENTED** [review]
We should do the similar work on https://github.com/rust-lang/rustc-dev-guide too before landing this one.

**Comment by lolbinarycat** [comment]
@LuuuXXX still interested in working on this?

**Comment by alex-semenyuk** [comment]
@LuuuXXX
Form wg-triage. Closed this PR due to inactivity. Feel free to reopen or raised new one. Thanks for your efforts.

**Pull request closed without merging**